pkg/pxc/app/binlogcollector/binlog-collector.go

```diff
@@ -49,7 +50,7 @@ func GetService(cr *api.PerconaXtraDBCluster) *corev1.Service {
 	}
 }
 
-func GetDeployment(cr *api.PerconaXtraDBCluster, initImage string) (appsv1.Deployment, error) {
+func GetDeployment(cr *api.PerconaXtraDBCluster, initImage string, existingSelector *metav1.LabelSelector) (appsv1.Deployment, error) {
```

Comment:
This function is not unit tested. Is it possible to add a basic unit test that mainly asserts that we have a valid deployment name, a valid number of containers and that the matching labels and the pod labels are the expected?